src/chainstate/stacks/db/transactions.rs

```diff
@@ -856,15 +856,32 @@ impl StacksChainState {
 
                 let (result, asset_map, events) = match contract_call_resp {
                     Ok((return_value, asset_map, events)) => {
-                        info!("Contract-call to {}.{:?} args {:?} returned {:?}", &contract_id, &contract_call.function_name, &contract_call.function_args, &return_value);
-                        info!("Contract-call to {}.{:?} args {:?} cost {:?}", &contract_id, &contract_call.function_name, &contract_call.function_args, &total_cost);
+                        info!("Contract-call successfully processed";
+                              "contract_name" => %contract_id,
+                              "function_name" => %contract_call.function_name,
+                              "function_args" => ?contract_call.function_args,
+                              "return_value" => %return_value,
+                              "cost" => ?total_cost);
                         Ok((return_value, asset_map, events))
                     },
                     Err(e) => {
                         match e {
                             // runtime errors are okay -- we just have an empty asset map
                             clarity_error::Interpreter(InterpreterError::Runtime(ref runtime_error, ref stack)) => {
-                                info!("Runtime error {:?} on contract-call {}.{:?} {:?}, stack trace {:?}", runtime_error, &contract_id, &contract_call.function_name, &contract_call.function_args, stack);
+                                info!("Contract-call processed with error";
+                                      "contract_name" => %contract_id,
+                                      "function_name" => %contract_call.function_name,
+                                      "function_args" => ?contract_call.function_args,
+                                      "error" => ?runtime_error,
+                                      "stack" => ?stack);
+                                Ok((Value::err_none(), AssetMap::new(), vec![]))
+                            },
+                            clarity_error::Interpreter(InterpreterError::ShortReturn(ref short_ret)) => {
```

Comment:
There aren't any other such `InterpreterError`s we should handle here, right?

Reply:
I don't think so -- the other error kinds are Unchecked (which should have been prevented through analysis) and Interpreter (which should not occur at all), both of which a miner should not be allowed to mine a block with.